homeassistant/components/lutron_caseta/button.py

```diff
@@ -0,0 +1,80 @@
+"""Support for pico and keypad buttons."""
+from __future__ import annotations
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import LutronCasetaDevice
+from .const import DOMAIN as CASETA_DOMAIN
+from .device_trigger import (
+    LEAP_TO_DEVICE_TYPE_SUBTYPE_MAP,
+    _lutron_model_to_device_type,
+)
+from .models import LutronCasetaData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lutron pico and keypad buttons."""
+    data: LutronCasetaData = hass.data[CASETA_DOMAIN][config_entry.entry_id]
+    bridge = data.bridge
+    button_devices = bridge.get_buttons()
+
+    async_add_entities(
+        LutronCasetaButton(button_devices[button_device_id], data)
+        for button_device_id in button_devices
+    )
+
+
+class LutronCasetaButton(LutronCasetaDevice, ButtonEntity):
+    """Representation of a Lutron pico and keypad button."""
+
+    def __init__(self, device, data):
+        """Init a button entity."""
+
+        super().__init__(device, data)
+        self._enabled_default = True
+
+        parent_device_info = data.device_info_by_device_id.get(device["parent_device"])
+
+        if not (device_name := device.get("device_name")):
+            # device name (button name) is missing, probably a caseta pico
+            # try to get the name using the button number from the triggers
+            # disable the button by default
+            self._enabled_default = False
+            keypad_device = self._smartbridge.get_devices()[device["parent_device"]]
+            button_numbers = LEAP_TO_DEVICE_TYPE_SUBTYPE_MAP.get(
+                _lutron_model_to_device_type(
+                    keypad_device["model"], keypad_device["type"]
+                ),
+                {},
+            )
+            device_name = button_numbers.get(
+                int(device["button_number"]),
+                f"button {device['button_number']}",
+            )
+
```

Comment:
Let's move this logic back to `async_setup_entry` and pass in the result to the constructor instead so the entity knows a little as possible about the implementation details of the library

Reply:
should I make a copy of the button_devices, and add the "device_name" key to it, so not changing the original library data?

    button_devices = bridge.get_buttons().copy()

```
    button_devices = bridge.get_buttons()

    async_add_entities(
        LutronCasetaButton(button_devices[button_device_id], data)
        for button_device_id in button_devices
    )
```

Reply:
don't see how I can use this simple loop if I'm going to pass in another variable with a different value for each button

```
    async_add_entities(
        LutronCasetaButton(button_devices[button_device_id], data)
        for button_device_id in button_devices
    )
```

Reply:
If I use the copy method:

```
async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up Lutron pico and keypad buttons."""
    data: LutronCasetaData = hass.data[CASETA_DOMAIN][config_entry.entry_id]
    bridge = data.bridge
    button_devices = bridge.get_buttons().copy()

    for button in button_devices:
        if not (device_name := button.get("device_name")):
            # device name (button name) is missing, probably a caseta pico
            # try to get the name using the button number from the triggers
            # disable the button by default

            keypad_device = bridge.get_devices()[button["parent_device"]]
            button_numbers = LEAP_TO_DEVICE_TYPE_SUBTYPE_MAP.get(
                _lutron_model_to_device_type(
                    keypad_device["model"], keypad_device["type"]
                ),
                {},
            )
            button_devices.update({
                "device_name": button_numbers.get(int(button["button_number"]),f"button {button['button_number']}"),
                "enabled_default": False,
            })

    async_add_entities(
        LutronCasetaButton(button_devices[button_device_id], data)
        for button_device_id in button_devices
    )
```

Reply:
I'll push up a suggestion shortly

Reply:
Untested but something like this should do it https://github.com/home-assistant/core/commit/63a7d41b59432832b7ff99a99140e8b0d34bc701

Reply:
Looks great.  Tested beautifully on my system.
Originally I had tried to use the DeviceInfo type in the last PR, but couldn't get the dr.async_get_or_create() to work, but I see separating out the config_entry_id does the trick.  

Thanks!